### PR TITLE
Vulkan: Swizzle and decode (some) textures directly

### PR DESCRIPTION
**Pull request by unknownbrackets**
This doesn't do the paletted formats directly, but it does the others.

Also, texture scaling now works properly.

-[Unknown]

### CODE REVIEW

File: GPU/Vulkan/TextureScalerVulkan.cpp [hunk 1]
```diff
@@ -21,38 +21,45 @@
...
-#include "GPU/Common/TextureScalerCommon.h"
-#include "GPU/Vulkan/TextureScalerVulkan.h"
+#include "Common/Vulkan/VulkanContext.h"
 #include "Common/ColorConv.h"
 #include "Common/Log.h"
 #include "Common/ThreadPools.h"
+#include "GPU/Common/TextureScalerCommon.h"
+#include "GPU/Vulkan/TextureScalerVulkan.h"
+
+// TODO: Share in TextureCacheVulkan.h?
```
Comment (hrydgard) [inline GPU/Vulkan/TextureScalerVulkan.cpp]: I prefer GPU/Vulkan/VulkanUtils.cpp/h (which may be renamed or split later) for "ppsspp-specific" things. Ideally, Common/Vulkan should not know about PPSSPP concepts.

File: GPU/Vulkan/TextureCacheVulkan.cpp [hunk 2]
```diff
@@ -1541,93 +1537,96 @@ void *TextureCacheVulkan::DecodeTextureLevel(GETextureFormat format, GEPaletteFo
...
-			finalBuf = tmpTexBuf16.data();
-			ConvertColors(finalBuf, texptr, dstFmt, bufw * h);
+			// Just a simple copy, we swizzle the color format.
+			for (int y = 0; y < h; ++y) {
+				memcpy(out + outPitch * y, texptr + bufw * sizeof(u16) * y, w * sizeof(u16));
+			}
+			finalBuf = out;
 		} else {
 			tmpTexBuf32.resize(std::max(bufw, w) * h);
 			UnswizzleFromMem(tmpTexBuf32.data(), texptr, bufw, h, 2);
```
Comment (hrydgard) [inline GPU/Vulkan/TextureCacheVulkan.cpp]: I'm sure you've thought of this, but UnswizzleFromMem could use an outPitch parameter - with that, we could avoid the memcpy loop. I'm fine with leaving that for later though, of course.

File: GPU/Vulkan/TextureScalerVulkan.cpp (hunk 1 above)
Comment (unknownbrackets) [inline GPU/Vulkan/TextureScalerVulkan.cpp]: Of course, I meant expose it in the TextureCacheVulkan header.  Maybe utils makes sense... it's just for the cases.

-[Unknown]

File: GPU/Vulkan/TextureCacheVulkan.cpp (hunk 2 above)
Comment (unknownbrackets) [inline GPU/Vulkan/TextureCacheVulkan.cpp]: Right, figured this was an easy first step, just to make sure it works and doesn't cause unexpected issues.

-[Unknown]

File: GPU/Vulkan/TextureScalerVulkan.cpp (hunk 1 above)
Comment (hrydgard) [inline GPU/Vulkan/TextureScalerVulkan.cpp]: Uh wait, I'm tired, I read it as VulkanImage.cpp for some reason. Never mind my comment.

**Pull request merged**